### PR TITLE
Magic link / Add auth origin payload

### LINKED ISSUE #471: Add "auth_origin" payload to the Magic Link token
### Overview
To determine the originator of sending a Magik Link to a user we need to save this information in the Magic Link token.

### Action Items

- [x] Create a JavaScript util class for string constants.
- [x] Create a constant `['LOG_IN', 'ONBOARDING', 'GOOGLE_ACCOUNT_CHECK']`.
- [x] Send `auth_origin = LOG_IN` when requesting the Magic Link for logging in.
- [x] Send `auth_origin = ONBOARDING` when requesting the Magic Link for checking out if the user email is valid during onboarding.
- [x] Save `auth_origin` into token on the backend.
- [x] Parse `auth_origin` from the Magic Link on the backend when the user opened it.
- [x] Send `auth_origin` back to the frontend in both cases of success or failure.

### PR DESCRIPTION
**Pull request by OlgaBilogurova**
Fixes #471 

@alex-anakin please, review especially backend changes I made for this issue.
I completed all action items from the task besides the last one:
"Send auth_origin back to the frontend in both cases of success or failure."
Currently, in case of failure, the backend sends status `res.sendStatus(400)` or `res.sendStatus(401)`. What changes do you want me to do there?

### CODE REVIEW

File: backend/controllers/user.controller.js [hunk 1]
```diff
@@ -162,6 +167,8 @@ UserController.verifySignIn = async function (req, res) {
 
   try {
     const payload = jwt.verify(token, CONFIG_AUTH.SECRET);
+    // TODO: remove console.log
+    console.log(payload);
```
Comment (alex-anakin) [inline backend/controllers/user.controller.js]: please remove it

**Review by alex-anakin — CHANGES_REQUESTED** [review]
Please make changes to the old client:
1. Send `LOG_IN` when requesting authorization.
2. Parse changed the magic link response properly.
3. Test logging in on the old client.

**Comment by peteplass** [comment]
Reviewed in meeting - would like update by 4/26

**Comment by OlgaBilogurova** [comment]
This week I'm going to make changes for this PR.

**Comment by OlgaBilogurova** [comment]
Update:
- Resolved comment
- Reworked `auth` logic for `AdminLogin` component for the old client to have login functionality working (for admin user role).
- Merged changes from `development `and have 1 backend test failing.

I'm going to refactor and rename this component, and probably extract API requests to a separate file.
Will do it this week.

**Review by alex-anakin — APPROVED** [review]
Looks good to me

**Pull request merged**